### PR TITLE
Modified main.py

### PR DESCRIPTION
**Pull request by muthuubalakan**
main.py modified with good understandable comments. 
Implemented cookie_check function for checking the valid cookie on every page user visiting.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,35 +1,77 @@
+''' Simple aiohttp login system with default user!:
+This system is built with aiohttp module.
+Required Python-3 or 3+
```
Comment (hgn) [inline main.py]: no, please add a shebang line, you also probably want to check a certain python version. asyncio was introduced in python x.y (google this). now you can add an check somewhere and if the required version is not sufficiently you can print a nice and helpful message and quit the execution.

File: main.py [hunk 2]
```diff
@@ -1,35 +1,77 @@
...
+'''
+
 from aiohttp import web
 
+# Customize the cookie name, value and cookie expire date
+COOKIE_NAME = "OldTamil"
+COOKIE_VALUE = 0
+COOKIE_LIFE_TIME = 24*60*60*30
```
Comment (hgn) [inline main.py]: always spaces between operators:

```
COOKIE_LIFE_TIME = 24 * 60 * 60 * 30
```

File: main.py [hunk 3]
```diff
@@ -1,35 +1,77 @@
...
 
-async def index(request):
-    cookie = request.cookies.get('default', None)
+# User credentials and here we could use more users in future.
+User ='Hippod'
```
Comment (hgn) [inline main.py]: no, it is global define, all global defines should be uppercase, like the other. Anyway, please ask yourself, is it cool when the style look not identical some lines down? Your answer should be: "no, it looks somehow shitty, i adjust the style". So User -> USER, probably use USERNAME, not just user. This is more exact. Always be so exact as possible if you name variables

File: main.py [hunk 4]
```diff
@@ -1,35 +1,77 @@
...
+Password = 'admin'
+
+# Define the required host and port
+HOST = '127.0.0.1'
+PORT = 8080
+
+''' cookie_check function will check for valid cookie on all pages!
+Return True if there is cookie with requested name.
+'''
+def cookie_check(request):
```
Comment (hgn) [inline main.py]: wait! Where do you comment a function and where do other comment functions? Do not invent your own style.

It is time that you should read this (referenced by pep8): https://www.python.org/dev/peps/pep-0257/

File: main.py [hunk 5]
```diff
@@ -1,35 +1,77 @@
...
+        # Normal test for checking the COOKIE_NAME
+        else:
+            print("COOKIE_NAME has been modified!.")
+    # Return False when there is no cookie at all.
+    else:
+        return False
+
+# The main page of the website
+async def index(request):
+    Check = cookie_check(request)
```
Comment (hgn) [inline main.py]: no, please read pep8, Check is not how you should name local variables

File: main.py [hunk 6]
```diff
@@ -1,35 +1,77 @@
...
+    if Check is True:
         page = open('site.html').read()
```
Comment (hgn) [inline main.py]: if think it is time to replace this construct: `open('site.html').read()`

What is the problem? Well, you do no error handling at all! What if the file is not available? The program crash! This should never happens!

Please extend and use (somehow, never tried it):

```
def prepare_file(filepath):
    if not os.isfile(filepath):
         print("internal server error, file not found: {}".format(filepath))
         return "<html>INTERNAL SERVER ERROR<html>"
    return open('site.html').read()
```

File: main.py [hunk 7]
```diff
@@ -1,35 +1,77 @@
...
-        if password == 'admin':
-            response = web.HTTPFound(location='/')
-            response.set_cookie('default', value=1, max_age=30*86400, path='/')
-            return response
-    else:
-        resp = web.HTTPFound(location='/loginpage')
-        return resp
+        form = await request.post()
+        username = form.get('username')
+        if username == User:
```
Comment (hgn) [inline main.py]: here, the problems comes into play, here `User` looks like a local variable! If you wrote it with UPPERCASE everybody knows instantly that you compare `username` with a global definition. Much more easy to follow the code.

File: main.py [hunk 8]
```diff
@@ -1,35 +1,77 @@
...
+            password = form.get('password')
+            if password == Password:
```
Comment (hgn) [inline main.py]: Please, I said that you should simplify the code, please check my old suggestion

**Review by hgn — CHANGES_REQUESTED** [review]

**Comment by hgn** [comment]
Beside the comments, the code looks far better then some days back! We are making progress! Some days and the code will look professional! :-) Thank you @muthuubalakan - good progress.

@lwrl please also check it again.

File: main.py [hunk 9]
```diff
@@ -1,77 +1,112 @@
...
-            return True
-        # Normal test for checking the COOKIE_NAME
-        else:
-            print("COOKIE_NAME has been modified!.")
-    # Return False when there is no cookie at all.
-    else:
+    if not cookie:
+        return False
+    if COOKIE_VALUE == 0:
+        return True
```
Comment (muthuubalakan) [inline main.py]: Here the function can check for cookie with name and value! but somehow, in future, the name and value can be changed.

File: main.py [hunk 10]
```diff
@@ -1,35 +1,112 @@
...
+def check_cookie(request):
+    """cookie_check!
+    This function will check for valid cookie on all pages!
+    Return True if there is valid cookie.
+
+    """
+    cookie = request.cookies.get(COOKIE_NAME, None)
+    if not cookie:
+        return False
+    if COOKIE_VALUE == 0:
```
Comment (hgn) [inline main.py:38]:
````
if cookie == COOKIE_VALUE:
      return True
return False
```

**Pull request merged**